content/doc/book/system-administration/reverse-proxy-configuration-with-jenkins.adoc

```diff
@@ -0,0 +1,1411 @@
+---
+layout: section
+---
+
+ifdef::backend-html5[]
+ifndef::env-github[:imagesdir: ../../resources/managing]
+:notitle:
+:description:
+:author:
+:email: jenkinsci-users@googlegroups.com
+:sectanchors:
+:toc: left
+endif::[]
+
+= Reverse Proxy Configuration with Jenkins
+
+A 'link:https://en.wikipedia.org/wiki/Reverse_proxy[reverse proxy]' allows an alternate HTTP or HTTPS provider to communicate with web browsers on behalf of Jenkins.
+The alternate provider may offer additional capabilities, like SSL encryption.
+The alternate provider may offload some work from Jenkins, like delivering static images.
+
+== General Guidelines
+
+Jenkins actively monitors reverse proxy configuration.
+Jenkins reports <<Jenkins says my reverse proxy setup is broken,"Your reverse proxy setup is broken">> when it detects a reverse proxy configuration problem.
+See the <<troubleshooting>> section if Jenkins is reporting that your reverse proxy setup is broken.
+
+=== Background
+
+Reverse proxies receive inbound HTTP requests and forward those requests to Jenkins.
+It receives the outbound HTTP response from Jenkins and forwards those requests to the original requester.
+A correctly configured reverse proxy rewrites *both* the HTTP request and the HTTP response.
+
+When HTTP request rewriting is misconfigured, pages won't be displayed at all.
+Refer to <<Configuration Examples,configuration examples>> if your reverse proxy is not displaying any Jenkins pages.
+
+A reverse proxy must handle the HTTP response by either rewriting the response or setting HTTP headers on the forwarded request.
+When HTTP response handling is misconfigured, Jenkins may fail to show updated information on a page or it may ignore changes submitted through web pages.
+See the <<troubleshooting>> section if Jenkins is reporting that your reverse proxy setup is broken or pages are not behaving as expected.
+
+== Configuration Examples
+
+Jenkins works with many different reverse proxies.
+This page provides examples for specific reverse proxies, though much of the information also applies to other reverse proxies.
+
+* <<Running Jenkins behind Apache>>
+* <<Running Jenkins behind Nginx>>
+* <<Running Jenkins behind HAProxy>>
+* <<Running Jenkins behind Squid>>
+* <<Running Jenkins behind IIS>>
+
+=== Running Jenkins behind Apache
+
+In situations where you have existing web sites on your server,
+you may find it useful to run Jenkins (or the servlet container
+that Jenkins runs in) behind Apache, so that you can bind Jenkins
+to the part of a bigger website that you may have.
+This section discusses some of the approaches for doing this.
+
+*Make sure that you change the Jenkins httpListenAddress from its
+default of 0.0.0.0 to 127.0.0.1 or any Apache-level restrictions can be
+easily bypassed by accessing the Jenkins port directly.*
+
```

Comment:
The Apache configuration sections that appear later in the document seem like they are each offering a different technique that might be used to configure Jenkins behind an Apache reverse proxy.  If that is correct, then I think these should be an introduction paragraph with navigation links that will allow the reader to choose their preferred technique.
```suggestion

There are several different alternatives to configure Jenkins with Apache.
Choose the technique that best meets your needs:

* <<mod_proxy>>
* <<mod_proxy with HTTPS>>
* <<mod_ajp/mod_proxy_ajp>>
* <<mod_proxy_ajp+SSL>>
* <<mod_rewrite>>

```

Double check that all the alternatives are included in the list.  I did not perform the detailed check.